packages/dapp/src/components/newsroom/NameAndAddress.tsx

```diff
@@ -42,13 +45,22 @@ export class NameAndAddress extends React.Component<StepProps, NameAndAddressSta
       </Collapsable>
     </StepStyled>)
   }
+
   private createNewsroom = async (): Promise<TwoStepEthTransaction<any>> => {
     const civil = getCivil();
     return civil.newsroomDeployTrusted(this.state.name);
   };
 
-  // private onNewsroomCreated = (result: any) => {
-  //   const address = result.address;
-  //   this.props.history.push("/mgmt/" + address);
-  // };
+  private onNewsroomCreated = async (result: any) => {
+    const civil = getCivil();
+    const user = civil.userAccount;
+    const newsroomWrapper = await result.getNewsroomWrapper();
+    this.props.dispatch!(addNewsroom(newsroomWrapper));
+    if (user && newsroomWrapper.data.owners.includes(user)) {
+      this.props.dispatch!(addUserNewsroom(result.address));
+      this.props.dispatch!(addOrUpdateUIState("newsroomMgmtCurrentAddress", result.address));
```

Comment:
Please see my comments in `src/reducers/ui.ts`!

Reply:
dang, I did this too but in a different way